kafka-rest/src/main/java/io/confluent/kafkarest/controllers/SchemaManagerImpl.java

```diff
@@ -55,31 +65,45 @@ public RegisteredSchema getSchema(
       Optional<String> rawSchema,
       boolean isKey) {
     // (subject|subjectNameStrategy)?, schemaId
+
+    if (!schemaRegistryEnabled && !format.isPresent()) {
```

Comment:
If schema registry is not configured, it should always throw, regardless of `format`. Even with `format`, we still need schema registry. We are supposed to insert the schema in SR before producing the message (the message is produced to Kafka with the SR id, not with the schema).

Also, I have a preference for having a separate class implementation that just throws, and doing the if in the module, but author's choice. As a rule of thumb, if the logic is based on something we know at start up time, instead of request time, it's better to do it in the module, IMO.